Lib/dataclasses.py

```diff
@@ -1235,13 +1230,38 @@ def _update_func_cell_for__class__(f, oldcls, newcls):
     # Fix the cell to point to the new class, if it's already pointing
     # at the old class.  I'm not convinced that the "is oldcls" test
     # is needed, but other than performance can't hurt.
-    closure = f.__closure__[idx]
-    if closure.cell_contents is oldcls:
-        closure.cell_contents = newcls
+    cell = f.__closure__[idx]
+    if cell.cell_contents is oldcls:
+        cell.cell_contents = newcls
         return True
     return False
 
 
+def _find_inner_functions(obj, _seen=None, _depth=0):
+    if _seen is None:
+        _seen = set()
+    if id(obj) in _seen:
+        return None
+    _seen.add(id(obj))
+
+    _depth += 1
+    # Normally just inspection of a descriptor object itself should be enough,
+    # and we should encounter the function as its attribute,
+    # but in case function was wrapped (e.g. functools.partition was used),
```

Comment:
```suggestion
    # Normally just an inspection of a descriptor object itself should be enough,
    # and we should encounter the function as its attribute,
    # but in case function was wrapped (e.g. functools.partial was used),
```